### PR TITLE
CheckBox: Test all configuration using TestCafe

### PR DESCRIPTION
**Pull request by jdvictoria**

### CODE REVIEW

**Comment by jdvictoria** [comment]
[CheckBox: test all configuration using testcafe](https://trello.com/c/hIfp1hVu/8397-checkbox-test-all-configuration-using-testcafe)

File: e2e/testcafe-devextreme/tests/editors/checkBox/common.ts [hunk 1]
```diff
@@ -47,3 +59,66 @@ fixture.disablePageReloads`CheckBox`
...
+}).before(async () => {
+  for (const iconSize of [20, undefined, 45]) {
+    for (const state of [
+      READONLY_STATE_CLASS,
+      DEFAULT_STATE_CLASS,
+      ACTIVE_STATE_CLASS,
+      HOVER_STATE_CLASS,
+      FOCUSED_STATE_CLASS,
+      DISABLED_STATE_CLASS,
+    ] as string[]
```
Comment (EugeniyKiyashko) [inline e2e/testcafe-devextreme/tests/editors/checkBox/common.ts]: Also 'dx-invalid' state is missed. It is also necessary to test the intersections of states. For example: checkbox can be in an invalid state and be focused

File: e2e/testcafe-devextreme/tests/editors/checkBox/common.ts [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
+
+const valueModes = [false, true, undefined];
 
 const CHECKBOX_CLASS = 'dx-checkbox';
+const READONLY_STATE_CLASS = 'dx-state-readonly';
+const DEFAULT_STATE_CLASS = '';
+const ACTIVE_STATE_CLASS = 'dx-state-active';
+const HOVER_STATE_CLASS = 'dx-state-hover';
+const FOCUSED_STATE_CLASS = 'dx-state-focused';
+const DISABLED_STATE_CLASS = 'dx-state-disabled';
```
Comment (EugeniyKiyashko) [inline e2e/testcafe-devextreme/tests/editors/checkBox/common.ts:23]: There are no tests for checkboxes with scaling applied. Something like here: https://github.com/EugeniyKiyashko/DevExtreme/blob/88ae219ba389a4e89473e9c437f2d3ef97119c7f/e2e/testcafe-devextreme/tests/editors/radioGroup/common.ts#L91-L92

File: e2e/testcafe-devextreme/tests/editors/checkBox/common.ts [hunk 3]
```diff
@@ -47,3 +59,66 @@ fixture.disablePageReloads`CheckBox`
...
+
+test('Checkbox configuration, different Checkbox icon sizes, states, value Modes', async (t) => {
+  const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+
+  await testScreenshot(t, takeScreenshot, 'CheckBox configurations different iconSizes.png', { element: '#container', shouldTestInCompact: true });
+
+  await testScreenshot(t, takeScreenshot, 'CheckBox configurations different iconSizes.png', { element: '#container', theme: getDarkThemeName() });
+
+  await t
+    .expect(compareResults.isValid())
```
Comment (EugeniyKiyashko) [inline e2e/testcafe-devextreme/tests/editors/checkBox/common.ts]: When resizing one of the checkboxes in the #container element, the size of the container will change. As a result, we will not see a difference in screenshots, as the size of the screenshot will be changed.

**Review by EugeniyKiyashko — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Review by AlexanderMoiseev — APPROVED** [review]

**Pull request merged**